src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -153,7 +154,7 @@ Copyright (c) .NET Foundation. All rights reserved.
                             _ComputePackageReferencePublish"
           BeforeTargets="CopyFilesToOutputDirectory"
           Condition="'$(GenerateDependencyFile)' == 'true'"
-          Inputs="$(ProjectAssetsFile);$(MSBuildAllProjects)"
+          Inputs="$(ProjectAssetsFile);$(ProjectAssetsCacheFile);$(MSBuildAllProjects)"
```

Comment:
I was about to point out that if DisablePackageAssetsCache is set, then the files will always be regenerated. As msbuild will log:

```
Building target "Building target "Build" completely. Input file "foo.assets.cache" does not exist." completely. Input file "foo.assets.cache" does not exist.
```

And say that it was a problem. But actually, I think it works out well. The only reason to disable the assets cache is as an escape hatch for a bug. Once you do that, incremental builds slow down considerably by design. So, the build would still be correct in all cases, it would just slow down even further. :)

Reply:
There might be an extreme edge case.

build with assets cache on
build again with assets cache off, change self contained.

I don't think it's worth trying to fix, though.

Reply:
I suppose it isn't that hard to fix, we could delete the assets cache file when assets cache is disabled. Still not sure if it matters.

Reply:
I think we can probably do without addressing that corner case with these changes.